### PR TITLE
Add method-level documentation for SAGE

### LINKED ISSUE #32: Add documentation for methods
## Description

Currently, block comments are present in the code, helping the user understand the related parts. However, a method-level documentation in the form of Python Docstrings is missing.

## Proposed solution

Document all the (non-helper) methods using Python Docstrings, in the following format:
image
image

### PR DESCRIPTION
**Pull request by jzelenjak**
Closes #32.

### CODE REVIEW

File: plotting.py [hunk 1]
```diff
@@ -135,17 +154,31 @@ def plot_episodes(frequencies, episodes, mcat):
         plt.plot(xax_end, yax, 'r', linestyle=(0, (5, 10)))
 
     plt.show()
-    return
 
 
 def _legend_without_duplicate_labels(ax, fontsize=10, loc='upper right'):
+    """
+    Creates a legend without duplicate labels.
```
Comment (jzelenjak) [inline plotting.py]: Well, I really don't know what else I can write here :)

File: plotting.py [hunk 2]
```diff
@@ -112,13 +124,20 @@ def plot_histogram(_team_alerts, _team_labels, experiment_name, suricata_summary
...
-    plt.legend([plot[0] for plot in plots], _team_labels)
+    plt.legend([plot[0] for plot in plots], team_labels)
     plt.tight_layout()
     plt.savefig('data_histogram-' + experiment_name + '.png')
     # plt.show()
 
 
 def plot_episodes(frequencies, episodes, mcat):
+    """
+    Plot the slopes based on the time, for a given (hyper)alert sequence (for an attacker-victim pair and mcat).
```
Comment (jzelenjak) [inline plotting.py]: Is this more or less the purpose of this method?

File: plotting.py [hunk 3]
```diff
@@ -135,17 +154,31 @@ def plot_episodes(frequencies, episodes, mcat):
...
+    """
     handles, labels = ax.get_legend_handles_labels()
     unique = [(h, l) for i, (h, l) in enumerate(zip(handles, labels)) if l not in labels[:i]]
     unique = sorted(unique, key=lambda x: x[1])
     ax.legend(*zip(*unique), loc=loc, fontsize=fontsize)
 
 
 def plot_alert_volume_per_episode(tid, attacker_victim, host_episodes, mcats):
+    """
+    Plots the alert volume per episode for the episodes of the given attacker-victim pair and the given mcat.
```
Comment (jzelenjak) [inline plotting.py]: Is this more or less the purpose of this method?

File: README.md [hunk 4]
```diff
@@ -48,7 +48,7 @@ Tip: in case you often use the same non-default values, you can create an alias
 
 ## First time use
 
-- In `sage.py`, set paths to `flexfringe/` executable, and `path_to_ini` variable.
+- In `sage.py`, set paths to `flexfringe/` executable, and `path_to_ini` variable (path to the `spdfa-config.ini` file, depending on whether you want to move this file to `FlexFringe/ini/` or keep it in the SAGE directory as it is).
```
Comment (azqanadeem) [inline README.md]: The additional comment is confusing. Better to give explicit options with commands, like e.g., Flexfringe is in /random/path, so we can set variable X to  /random/path.

File: ag_generation.py [hunk 5]
```diff
@@ -3,8 +3,14 @@
 from signatures.mappings import macro_inv, micro, micro2macro, verbose_micro
 
 
-# Translate technical nodes to human-readable
 def _translate(label, root=False):
+    """
+    Translates the node label into a more human-readable version.
+
+    @param label: the label of the node (`mcat|mserv`, where mcat is an integer)
```
Comment (azqanadeem) [inline ag_generation.py]: mcat is an integer mapped from a dictionary? State the dictionary name as well for clarity.

File: plotting.py (hunk 2 above)
Comment (azqanadeem) [reply plotting.py]: Plot the rate of alerts over time, for a given hyperalert sequence of specific mcat and belonging to an attacker-victim pair

File: plotting.py (hunk 1 above)
Comment (azqanadeem) [reply plotting.py]: Removes redundant legend handles.

File: plotting.py (hunk 3 above)
Comment (azqanadeem) [reply plotting.py]: No, it's more like it plots the episodes over time for each attacker victim pair. It shows a global view of the different attacker actions over time

File: plotting.py (hunk 1 above)
Comment (jzelenjak) [reply plotting.py]: Updated

File: plotting.py (hunk 2 above)
Comment (jzelenjak) [reply plotting.py]: Updated

File: plotting.py (hunk 3 above)
Comment (jzelenjak) [reply plotting.py]: Updated

File: README.md (hunk 4 above)
Comment (jzelenjak) [reply README.md]: Is it now better?

File: ag_generation.py (hunk 5 above)
Comment (jzelenjak) [reply ag_generation.py]: Right, this was a mistake from my side, sorry. `mcat` was actually a string. Thanks for letting me know! I have updated the doc

**Review by azqanadeem — APPROVED** [review]

**Pull request merged**